### PR TITLE
DOC: Change plot style to matplotlib decault from ggplot

### PR DESCRIPTION
**Pull request by topper-123**
A proposal to change plot style in the docs to ``seaborn`` from ``ggplot``.

See also #17423.

### CODE REVIEW

**Comment by jreback** [comment]
can you post a screen shot of the change (just a single page is fine, just something for people to look)

**Comment by topper-123** [comment]
Some plot files:
image

image


image

image

image

image

File: doc/source/visualization.rst [hunk 1]
```diff
@@ -1049,6 +1049,46 @@ be colored differently.
 Plot Formatting
 ---------------
 
+Setting the plot style
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+From version 1.5 and up, matplotlib offers a range of preconfigured plotting style. Setting these
```
Comment (jreback) [inline doc/source/visualization.rst]: can you point to the mpl docs here as well.

File: doc/source/visualization.rst [hunk 2]
```diff
@@ -1049,6 +1049,46 @@ be colored differently.
...
+these styles can be used to easily give the plots have the general look that you want.
+Setting the can style is as easy as calling ``matplotlib.style.use(my_plot_style)`` before
+creating your plot.
+
```
Comment (jreback) [inline doc/source/visualization.rst]: I don't think the examples of plot style are necessary here, again just point to the mpl docs.

**Comment by topper-123** [comment]
I couldn't find a suitable page where the built-in styles are demonstarted. But it's easy enough to do yourself, so I just deleted the examples.

**Comment by jreback** [comment]
lgtm. @TomAugspurger

**Comment by jorisvandenbossche** [comment]
We can use whatever version of matplotlib we want to built the docs, and I suppose we will typically use the latest. Given that, we can actually also use the new improved default matplotlib 2.0 style, instead of taking a custom one. 
My personal preference would be to just use the default matplotlib style now it has improved.

File: doc/source/visualization.rst [hunk 3]
```diff
@@ -1049,6 +1049,18 @@ be colored differently.
 Plot Formatting
 ---------------
 
+Setting the plot style
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+From version 1.5 and up, matplotlib offers a range of preconfigured plotting style. Setting the
```
Comment (jorisvandenbossche) [inline doc/source/visualization.rst]: style -> styles

File: doc/source/visualization.rst [hunk 4]
```diff
@@ -1049,6 +1049,18 @@ be colored differently.
...
+style can be used to easily give the plots the general look that you want.
```
Comment (jorisvandenbossche) [inline doc/source/visualization.rst]: the plots -> plots

**Review by jorisvandenbossche — COMMENTED** [review]
minor language comments.

**Comment by topper-123** [comment]
Language changes from @jorisvandenbossche implenented.

**Comment by bashtage** [comment]
MPL 2 isn't bad but still seaborn is nicer looking.

**Comment by jreback** [comment]
any objections to this. lgtm.

**Comment by topper-123** [comment]
I propose just merging this in. If anyone got an different view point, it's easy to make a new pull request.

**Comment by TomAugspurger** [comment]
Time for a poll!

1. ggplot (this PR)
2. seaborn (current)
3. matplotlib 2.0

Voting closes in 48 hours ;)

---

I vote for matplotlib 2.0

**Comment by bashtage** [comment]
Preferential voting n case of a tie  🥇 

2 > 3 > 1

FWIW I think the current is ggplot, and this PR is proposing to use seaborn

**Comment by topper-123** [comment]
2 > 3 > 1.

**Comment by jorisvandenbossche** [comment]
Any other votes ?

**Comment by jreback** [comment]
can someone post a comparison on 1, 2, 3 (for a single plot) here.

**Comment by jorisvandenbossche** [comment]
First example from the docs:

```
In [2]: ts = pd.Series(np.random.randn(1000), index=pd.date_range('1/1/2000', periods=1000))

In [3]: ts = ts.cumsum()

In [4]: ts.plot()
Out[4]: <matplotlib.axes._subplots.AxesSubplot at 0x1359f45f8>
```

```
In [11]: for style in ['ggplot', 'seaborn', 'default' ]:
    ...:     with plt.style.context(style):
    ...:         fig, ax = plt.subplots()
    ...:         ts.plot(ax=ax)
    ...:         fig.savefig("test_{}.png".format(style))
```

test_ggplot
test_seaborn
test_default

**Comment by jorisvandenbossche** [comment]
BTW, the reason I vote for 3 (matplotlib default) is not because I find it the nicest style (in that case I would actually vote for ggplot I think), but because I think it is good enough and it is the best option for our docs. When people run examples from the docs, they will most likely see the default style, and by using the default style in our docs, it will most likely match with what they see.

**Comment by jreback** [comment]
ok i can be on board with

3 > 2 > 1

**Comment by topper-123** [comment]
ok, I've changed it, so the plot style is not, which means default style for mpl > 2. The line is commented out rather than deleted, so its easy to experiment with different styles by doing find/replace.

**Comment by jorisvandenbossche** [comment]
@topper-123 Merged! 
As a follow-up, it would be good to check the dev docs if everything is looking more or less decent (once it is built, you should see the changes here: http://pandas-docs.github.io/pandas-docs-travis/visualization.html)

**Pull request merged**